src/deploy/NVA_build/noobaa_core.yaml

```diff
@@ -32,6 +32,12 @@ rules:
     verbs:
       - get
       - list
+  - apiGroups:
+    - authentication.k8s.io
+    resources:
+    - tokenreviews
```

Comment:
I am sorry but I must disagree. 
This is tested and working on OpenShift 4.1. 

TokenReviews are clustered scope but there isn't any current issue in binding any API access to any Role in k8s. 
So what I did here is to allow the NooBaa service account access to the toke reviews API and it doesn't really matter if the API is servicing cluster scoped resources or namespace scoped resources.

The reason we did not use a ClusterRoleBinding is that it mandates the use of a namespace in the subject section. The namespace (in which NooBaa be deployed at) is not determined until deploy time so it prevents me from using any ClusterRoleBinding inside the NooBaa YAML.

Please note that we support multiple NooBaa deployments on the same cluster under different namespaces and the deployment namespace is given by the deploying agent (usually an admin).

Reply:
> TokenReviews are clustered scope but there isn't any current issue in binding any API access to any Role in k8s.

You have a misunderstanding about how RBAC works.  You can create roles and bindings with literally anything in them.  That does not mean they actually do what you expect.

RBAC prevents things that live inside of a namespace from granting access at the cluster scope.

> So what I did here is to allow the NooBaa service account access to the toke reviews API and it doesn't really matter if the API is servicing cluster scoped resources or namespace scoped resources.

No, you granted it at the namespace scope which does nothing.  A namespaced token review API does not exist.

```
$ oc api-resources --api-group authentication.k8s.io
NAME           SHORTNAMES   APIGROUP                NAMESPACED   KIND
tokenreviews                authentication.k8s.io   false        TokenReview
```

> The reason we did not use a ClusterRoleBinding is that it mandates the use of a namespace in the subject section. The namespace (in which NooBaa be deployed at) is not determined until deploy time so it prevents me from using any ClusterRoleBinding inside the NooBaa YAML.

And you will have to update your scripts to tolerate that.

Reply:
@enj We are calling tokenreview on user login from within our web server, is it valid to request a cluster-role for tokenreview for our service-account which is namespaced?

@nb-ohad The operator will make it simple. But for the script it can just create this cluster-role + cluster-role-binding inline in the script after creating the yaml, and add to our service-account the permission to use tokenreview api in addition to the normal role+binding.

Reply:
> @enj We are calling tokenreview on user login from within our web server, is it valid to request a cluster-role for tokenreview for our service-account which is namespaced?

Yes.  For example, basically all kube controllers run as service accounts in particular namespaces.  As you can imagine, many of these access cluster scoped resources.